Sources/OutlineView/OutlineViewDelegate.swift

```diff
@@ -102,27 +102,36 @@ where Data.Element: Identifiable {
     func outlineViewItemDidExpand(_ notification: Notification) {
         let outlineView = notification.object as! NSOutlineView
         if outlineView.selectedRow == -1 {
-            selectRow(for: selectedItem, in: outlineView)
+            selectRow(for: selectedItems, in: outlineView)
         }
     }
 
     func outlineViewSelectionDidChange(_ notification: Notification) {
         let outlineView = notification.object as! NSOutlineView
-        if outlineView.selectedRow != -1 {
-            let newSelection = outlineView.item(atRow: outlineView.selectedRow).map(typedItem)
-            if selectedItem?.id != newSelection?.id {
-                selectedItem = newSelection
-                selectionChanged(selectedItem?.value)
+        let selectedRowIndexes = outlineView.selectedRowIndexes
```

Comment:
To get selectionChanged(xxx) to be called when I chose a new outline row I had to comment out some of your code. The first check for isEmpty shouldn't be needed because selectionChanged(emptySet) still needs to be called if you unselect every row. Also returning early because selectedItems and newSelection counts are same caused single selecting new items to not call selectionChanged(xxx) too. Also wasn't sure why the if selectedItems.allSatisfy call was there too?
Current code I'm using:
func outlineViewSelectionDidChange(_ notification: Notification) {
        let outlineView = notification.object as! NSOutlineView
        let selectedRowIndexes = outlineView.selectedRowIndexes
        
#warning("Check with original author why commented out code was needed")
        //if !selectedRowIndexes.isEmpty {
            let newSelections = selectedRowIndexes.compactMap {
                outlineView.item(atRow: $0).map(typedItem)
            }
            
//            if selectedItems.count == newSelections.count {
//                return
//            }
            
            //if selectedItems.allSatisfy({ newSelections.contains($0) }) {
                selectedItems = newSelections
                selectionChanged(Set(selectedItems.map(\.value)))
            //}
        //}
    }